File: PCWeb/src/types/etl/datasource.ts
/**
 * 数据源类型定义
 */

/**
 * 数据源类型
 */
export type DatasourceType = 'mysql' | 'postgresql' | 'oracle' | 'sqlserver' | 'clickhouse' | 'mongodb' | 'redis' | 'elasticsearch' | 'http_api' | 'ftp' | 'sftp' | 'hive'

/**
 * 数据源类型枚举值（用于常量引用）
 */
export const DatasourceType = {
  MYSQL: 'mysql' as DatasourceType,
  POSTGRESQL: 'postgresql' as DatasourceType,
  ORACLE: 'oracle' as DatasourceType,
  SQLSERVER: 'sqlserver' as DatasourceType,
  CLICKHOUSE: 'clickhouse' as DatasourceType,
  MONGODB: 'mongodb' as DatasourceType,
  REDIS: 'redis' as DatasourceType,
  ELASTICSEARCH: 'elasticsearch' as DatasourceType,
  HTTP_API: 'http_api' as DatasourceType,
  FTP: 'ftp' as DatasourceType,
  SFTP: 'sftp' as DatasourceType,
  HIVE: 'hive' as DatasourceType,
}

/**
 * 数据源状态
 */
export type DatasourceStatus = 'connected' | 'disconnected' | 'error'

/**
 * 数据源状态枚举值
 */
export const DatasourceStatus = {
  ACTIVE: 'connected' as DatasourceStatus,
  INACTIVE: 'disconnected' as DatasourceStatus,
  ERROR: 'error' as DatasourceStatus,
}

/**
 * 数据源信息
 */
export interface DataSource {
  id: string
  name: string
  type: DatasourceType
  host?: string
  port?: number
  database?: string
  username?: string
  status: DatasourceStatus
  lastConnectionTime?: string
  description?: string
  createTime?: string
  updateTime?: string
  creatorId?: string
  creatorName?: string
  /** 扩展配置（用于非数据库类型的数据源） */
  config?: Record<string, any>
}

/**
 * 数据源类型别名
 */
export type Datasource = DataSource

/**
 * 数据源列表查询参数
 */
export interface DataSourceListParams {
  pageIndex?: number
  pageSize?: number
  name?: string
  type?: DatasourceType
}

/**
 * 创建数据源参数
 */
export interface CreateDataSourceParams {
  name: string
  type: DatasourceType
  host: string
  port: number
  database: string
  username: string
  password: string
  description?: string
}

/**
 * 更新数据源参数
 */
export interface UpdateDataSourceParams {
  id: string
  name?: string
  host?: string
  port?: number
  database?: string
  username?: string
  password?: string
  description?: string
}

/**
 * 数据库类型信息
 */
export interface DbTypeInfo {
  code: string
  name: string
  defaultPort: number
}

/**
 * 测试连接结果
 */
export interface TestConnectionResult {
  success: boolean
  message: string
  serverVersion?: string
  connectionTime?: number
}

File: EasyWechatWeb/EasyWeChatModels/Entitys/Screen/Datasource.cs
using SqlSugar;

namespace EasyWeChatModels.Entitys;

/// <summary>
/// 数据源实体类
/// </summary>
/// <remarks>
/// 用于存储外部数据库连接信息，支持多种数据库类型
/// </remarks>
[SugarTable("Datasource", "数据源表")]
public class Datasource
{
    /// <summary>
    /// 数据源ID（主键）
    /// </summary>
    [SugarColumn(IsPrimaryKey = true)]
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>
    /// 数据源名称
    /// </summary>
    [SugarColumn(Length = 100, ColumnDescription = "数据源名称")]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// 数据源类型
    /// </summary>
    /// <remarks>
    /// 支持: mysql, postgresql, oracle, sqlserver, clickhouse
    /// </remarks>
    [SugarColumn(Length = 20, ColumnDescription = "数据源类型")]
    public string Type { get; set; } = "mysql";

    /// <summary>
    /// 主机地址
    /// </summary>
    [SugarColumn(Length = 100, ColumnDescription = "主机地址")]
    public string Host { get; set; } = string.Empty;

    /// <summary>
    /// 端口
    /// </summary>
    [SugarColumn(ColumnDescription = "端口")]
    public int Port { get; set; } = 3306;

    /// <summary>
    /// 数据库名称
    /// </summary>
    [SugarColumn(Length = 100, ColumnDescription = "数据库")]
    public string Database { get; set; } = string.Empty;

    /// <summary>
    /// 用户名
    /// </summary>
    [SugarColumn(Length = 50, ColumnDescription = "用户名")]
    public string Username { get; set; } = string.Empty;

    /// <summary>
    /// 密码
    /// </summary>
    [SugarColumn(Length = 100, ColumnDescription = "密码")]
    public string Password { get; set; } = string.Empty;

    /// <summary>
    /// 连接状态
    /// </summary>
    /// <remarks>
    /// 状态值: connected, disconnected, error
    /// </remarks>
    [SugarColumn(Length = 20, ColumnDescription = "连接状态")]
    public string Status { get; set; } = "disconnected";

    /// <summary>
    /// 最后连接时间
    /// </summary>
    [SugarColumn(IsNullable = true, ColumnDescription = "最后连接时间")]
    public DateTime? LastConnectionTime { get; set; }

    /// <summary>
    /// 描述
    /// </summary>
    [SugarColumn(Length = 500, IsNullable = true, ColumnDescription = "描述")]
    public string? Description { get; set; }

    /// <summary>
    /// 创建人ID
    /// </summary>
    [SugarColumn(ColumnDescription = "创建人ID")]
    public Guid CreatorId { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    [SugarColumn(ColumnDescription = "创建时间")]
    public DateTime CreateTime { get; set; } = DateTime.Now;

    /// <summary>
    /// 更新时间
    /// </summary>
    [SugarColumn(IsNullable = true, ColumnDescription = "更新时间")]
    public DateTime? UpdateTime { get; set; }
}